go/test/endtoend/onlineddl/vrepl_suite/onlineddl_vrepl_suite_test.go

```diff
@@ -0,0 +1,376 @@
+/*
+Copyright 2019 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vreplsuite
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+
+	"vitess.io/vitess/go/mysql"
+	"vitess.io/vitess/go/sqltypes"
+	"vitess.io/vitess/go/vt/schema"
+
+	"vitess.io/vitess/go/test/endtoend/cluster"
+	"vitess.io/vitess/go/test/endtoend/onlineddl"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	clusterInstance         *cluster.LocalProcessCluster
+	vtParams                mysql.ConnParams
+	evaluatedMysqlParams    *mysql.ConnParams
+	ddlStrategy             = "online -skip-topo -vreplication-test-suite"
+	executeStrategy         = "vtgate"
+	waitForMigrationTimeout = 20 * time.Second
+
+	hostname              = "localhost"
+	keyspaceName          = "ks"
+	cell                  = "zone1"
+	schemaChangeDirectory = ""
+	tableName             = `onlineddl_test`
+	beforeTableName       = `onlineddl_test_before`
+	afterTableName        = `onlineddl_test_after`
+	eventName             = `onlineddl_test`
+)
+
+const (
+	testDataPath   = "testdata"
+	defaultSQLMode = "ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION"
+)
+
+func TestMain(m *testing.M) {
+	defer cluster.PanicHandler(nil)
+	flag.Parse()
+
+	exitcode, err := func() (int, error) {
+		clusterInstance = cluster.NewCluster(cell, hostname)
+		schemaChangeDirectory = path.Join("/tmp", fmt.Sprintf("schema_change_dir_%d", clusterInstance.GetAndReserveTabletUID()))
```

Comment:
`ClusterInstance` has a `TmpDirectory` parameter (based off of VTDATAROOT). Maybe using that here might be more consistent with the other tests. I have found that helps while debugging: if you comment out the teardown temporarily all transient files are then in the same root dir.

Reply:
Actually, I inherited this `schemaChangeDirectory` variable, and am unsure how Vitess even uses it. Having said that, there's a problem with using suggested `TmpDirectory`:

- `TmpDirectory` is only set in `StartTopo()`
- `StartTopo()` requires that `VtctldExtraArgs` are set
- `VtctldExtraArgs` rely on `schemaChangeDirectory`

so if we introduce `schemaChangeDirectory  = cluster.TmpDirectory` there's a dependency loop.

I have no strong opinions (and really no opinions at all) on how this should be resolved. Even after reviewing the code I don't understand how `SchemaChangeDirName` is being used: it's used by `LocalController` -- I don't really understand what LocalController is?

Reply:
I'm gonna merge this PR with the nit still outstanding, and resolve `TmpDirectory` in a future PR :pray: